components/wb-suggest/suggest.js

```diff
@@ -0,0 +1,279 @@
+		input.value = input.value; // eslint-disable-line
+	},
+
+	// Initiate suggestion from received JSON file
+	//
+	// suggestion: JSON string array that was fetched
+	// this: the datalist reference
+	initSuggestion = function( suggestions ) {
+
+		// Attach the JSON list to the datalist element
+		this.dataset.wbSuggestions = JSON.stringify( suggestions );
+
+		// Remove the reference as it not needed anymore
+		delete this.dataset.wbSuggest;
+
+		// Add the suggested options
+		addDataListOptions.call( this, document.querySelector( "[list=" + this.id + "]" ).value );
+	},
+
+	inputFocus = function( event ) {
+
+		var target = event.target,
+			datalist = document.getElementById( target.getAttribute( "list" ) );
+
+		// IE11 add support for string normalization
+		Modernizr.load( {
+			test: Modernizr.stringnormalize,
+			nope: [
+				"site!deps/unorm" + wb.getMode() + ".js"
+			]
+		} );
+
+		// Load the JSON file
+		$( datalist ).trigger( {
+			type: "json-fetch.wb",
+			fetch: {
+				url: datalist.dataset.wbSuggest
+			}
+		} );
+	},
+
+	inputChange = function( event ) {
+		var target = event.target,
+			datalistElm = document.getElementById( target.getAttribute( "list" ) ),
+			query = event.target.value,
+			which = event.which;
+
+		if ( wait ) {
+			clearTimeout( wait );
+		}
+
+		switch ( event.type ) {
+		case "change":
+			wait = setTimeout( addDataListOptions.bind( datalistElm, query ), waitInterval );
+			break;
+		case "keyup":
+			if ( !( event.ctrlKey || event.altKey || event.metaKey ) ) {
+
+				// Backspace, Spacebar, a - z keys, 0 - 9 keys punctuation, and symbols
+				if ( which === 8 ||  which === 32 || ( which > 47 && which < 91 ) ||
+					( which > 95 && which < 112 ) || ( which > 159 && which < 177 ) ||
+					( which > 187 && which < 223 ) ) {
+
+					wait = setTimeout( addDataListOptions.bind( datalistElm, query ), waitInterval );
+				}
+			}
+		}
+	},
+
+	init = function( event ) {
+
+		var elm = wb.init( event, componentName, selector ),
+			inputSelector = "[list=" + event.target.id + "]";
+
+		if ( elm ) {
+			Modernizr.addTest( "stringnormalize", "normalize" in String );
+
+			//
+			// Add the trigger to load the JSON on when those input get focus.
+			// This avoid to download the JSON when it is not needed by the input
+			//
+			$document.one( "focus", inputSelector, inputFocus );
+
+			// Refresh the suggestion option based from user input if there is a limit set
+			if ( elm.dataset.wbLimit || elm.dataset.wbFilterType ) {
+				$document.on( "change keyup", inputSelector, inputChange );
+			}
+
+			// Initialization has completed
+			wb.ready( $( elm ), componentName );
+		}
+	};
+
+// Bind the init event of the plugin
+$document.on( "timerpoke.wb " + initEvent + " " + jsonFetched, selector, function( event ) {
+
+	var eventType = event.type,
+		eventTarget = event.target,
+		eventCurrentTarget = event.currentTarget;
+
+	if ( eventCurrentTarget === eventTarget ) {
```

Comment:
You could have used `event.currentTarget` directly and not create the `eventCurrentTarget` variable since it is only used once.

Reply:
Yes

Reply:
Fixed